lib/runCall.js

```diff
@@ -59,111 +61,287 @@ module.exports = function (opts, cb) {
   var isStatic = opts.static || false
   var salt = opts.salt || null
   var storageReader = opts.storageReader || new StorageReader(stateManager)
-
+  var results = {}
   txValue = new BN(txValue)
-
+  var client = new net.Socket()
   const kevmHost = '127.0.0.1'
   const kevmPort = 8080
+  var logList = []
 
-  var client = new net.Socket()
+  async.series([checkpointState, runClient], commitState)
 
-  client.connect(kevmPort, kevmHost, function () {
-    console.log('CONNECTED TO: ' + kevmHost + ':' + kevmPort)
-    var hello = createHello()
-    client.write(hello)
-    var callCtx = createCallContext()
-    client.write(callCtx)
-  })
-
-  // Add a 'data' event handler for the client socket
-  // data is what the server sent to this socket
-  client.on('data', function (data) {
-    try {
-      var query = msg_pb.VMQuery.deserializeBinary(data)
-      switch (query.getQueryCase()) {
-        case GETACCOUNT: {
-          console.log('GETACCOUNT', data)
-          computeGetAccount(query)
-          break
-        }
-        case GETSTORAGEDATA: {
-          console.log('GETSTORAGEDATA', data)
-          computeGetStorageData(query)
-          break
-        }
-        case GETCODE: {
-          console.log('GETCODE', data)
-          // TODO
-          break
+  function checkpointState (_callback) {
+    stateManager.checkpoint(_callback)
+  }
+
+  function runClient (_callback) {
+    // Add a 'data' event handler for the client socket
+    // data is what the server sent to this socket
+    client.on('data', function (data) {
+      try {
+        var query = msg_pb.VMQuery.deserializeBinary(data)
+        switch (query.getQueryCase()) {
+          case GETACCOUNT: {
+            computeGetAccount(query)
+            break
+          }
+          case GETSTORAGEDATA: {
+            computeGetStorageData(query)
+            break
+          }
+          case GETCODE: {
+            computeGetCode(query)
+            break
+          }
+          case GETBLOCKHASH: {
+            // TODO
+            break
+          }
+          case CALLRESULT: {
+            console.log('CALLRESULT1')
+            client.destroy()
+            computeCallResult(query, _callback)
+            break
+          }
+          case QUERY_NOT_SET: {
+            console.log('QUERYNOTSET')
+            break
+          }
         }
-        case GETBLOCKHASH: {
-          console.log('GETBLOCKHASH', data)
-          // TODO
-          break
+      } catch (err) {
```

Comment:
Let's localize this `catch` block to where the `try` is which is throwing the assertion error.

Reply:
Moved the `switch` statement outside of the `try` block.